pokemon-trivia/script.js

```diff
@@ -0,0 +1,84 @@
+// create a custom element (parameter 1), adding a className (parameter 2) and an innerText (parameter 3)
+function createCustomElement(element, className, innerText) {
+  const e = document.createElement(element);
+  e.className = className;
+  e.innerText = innerText;
+  return e;
+}
+
+// random number must be from 1 to 386 (3rd generation)
+function randomNumber() {
+  return Math.ceil(Math.random()*386);
+}
+
+// creates an array with the name and image of each pokemon
+async function pokemonNameAndImage() {
+  const pokemonId = randomNumber();
+  const pokemon = await fetch(`https://pokeapi.co/api/v2/pokemon/${pokemonId}`).then((res) => res.json());
+  const { species: { name: pokemonName }, sprites: { other: { dream_world: { front_default: pokemonImage } } } } = pokemon;
+  return [ pokemonName, pokemonImage ]
+}
+
+// create an arrays with 4 pokemons and images
+async function arrayOfAnswers() {
+  let pokemonArray = []
+  for (let i = 0; i < 4; i += 1) {
+    const pokemon = await pokemonNameAndImage();
+    pokemonArray.push(pokemon);
+  }
+  return pokemonArray;
+}
+
+function resetPage() {
+  document.location.reload(true);
+}
+
+// checks if the name inside the button corresponds to the image
+function checkResult() {
+  const clickedPokemon = this.firstChild.innerText.toLowerCase();
+  const correctPokemon = document.querySelector('.pokemonAnswer').innerText;
+  if (clickedPokemon !== correctPokemon) {
+    this.className = 'option wrong';
+    this.removeEventListener('click', checkResult);
+  }
+  if (clickedPokemon === correctPokemon) {
+    const modal = document.querySelector('.correct');
+    const again = document.querySelector('.again');
+    again.addEventListener('click', resetPage)
+    modal.className = 'correct'
+  }
+  console.log(clickedPokemon);
+  console.log(correctPokemon);
+}
+
+// create button and p for each pokemon in the array above
+function createOptions(pokemonArray) {
+  pokemonArray.forEach((pokemon) => {
+    const newButton = createCustomElement('button', 'option', '');
+    newButton.addEventListener('click', checkResult)
```

Comment:
Algo que poderia eliminar a necessidade de utilizar o event (ou this) na sua função checkResult, seria passar no próprio listener o nome do pokemon dessa forma:
``` suggestion
    newButton.addEventListener('click', () => checkResult(pokemon))
```
não que seja necessário, mas poderia ser útil caso não fosse possível pegar o nome do pokemon com o event